packages/backend-function/src/function_env_type_generator.test.ts

```diff
@@ -0,0 +1,59 @@
+import { describe, it, mock } from 'node:test';
+import fs from 'fs';
+import { FunctionEnvironmentTypeGenerator } from './function_env_type_generator.js';
+import assert from 'assert';
+import path from 'path';
+import { pathToFileURL } from 'url';
+
+void describe('FunctionEnvironmentTypeGenerator', () => {
+  void it('generates a type definition file', () => {
+    const fdCloseMock = mock.fn();
+    const fsOpenSyncMock = mock.method(fs, 'openSync');
+    const fsMkdirSyncMock = mock.method(fs, 'mkdirSync', () => null);
+    const fsWriteFileSyncMock = mock.method(fs, 'writeFileSync', () => null);
+    fsOpenSyncMock.mock.mockImplementation(() => {
+      return {
+        close: fdCloseMock,
+      };
+    });
+    const functionEnvironmentTypeGenerator =
+      new FunctionEnvironmentTypeGenerator(
+        'testFunction',
+        './test/function/handler.ts'
+      );
+    const sampleStaticEnv = '_HANDLER: string;';
+
+    functionEnvironmentTypeGenerator.generateTypeDefFile();
+
+    assert.equal(fsMkdirSyncMock.mock.callCount(), 1);
+    // assert type definition file path
+    assert.equal(
+      fsWriteFileSyncMock.mock.calls[0].arguments[0],
+      './test/function/amplify/testFunction_env.ts'
+    );
+    assert.ok(
+      fsWriteFileSyncMock.mock.calls[0].arguments[1]
+        ?.toString()
+        .includes(sampleStaticEnv)
+    );
+
+    mock.restoreAll();
+  });
+
+  void it('generated type definition file has valid syntax', async () => {
+    const targetDirectory = await fs.promises.mkdtemp('func_env_type_gen_test');
```

Comment:
import from `fs/promises` instead

Reply:
I was using `fs` since the test above was mocking `sync` methods, unless we are ok with additional import of `fs/promises`

Reply:
Yeah the additional import should be fine